frontend-react/src/components/Table/Table.tsx

```diff
@@ -244,13 +255,19 @@ const Table = ({
         );
     };
 
-    const DatasetActionButton = ({ label, method }: DatasetAction) => {
-        return (
-            <Button type={"button"} onClick={() => method()}>
-                {label}
-            </Button>
-        );
-    };
+    // this button will be placed above the rendered table and on `click` will run an arbitrary function
+    // passed in from the Table's parent, or an addRow function defined by the Table.
+    // in order to avoid problems around timing, takes a `disabled` prop.
+    // TODO: split this out of Table component
+    const DatasetActionButton = ({
+        label,
+        method = () => {},
```

Comment:
Would it make sense to default to `addRow` here instead of 238, or would it not default if `method` came through as undefined from `datasetAction.method`?

Reply:
As things stand I can see it going either way. In the end though, I think this component should live outside of the Table component, where it wouldn't have access to `addRow`, so I think doing it this way makes more sense from that perspective.